tools/amd_build/pyHIPIFY/hipify-python.py

```diff
@@ -0,0 +1,967 @@
+        print("The output folder already exists.")
+        return
+
+    # If no output directory, provide a default one.
+    if args.output_directory is "":
+        args.project_directory = args.project_directory[0:-
+                                                        1] if args.project_directory.endswith("/") else args.project_directory
+        args.output_directory = args.project_directory + "_amd"
+
+    # Copy from project directory to output directory if not done already.
+    if not os.path.exists(args.output_directory):
+        shutil.copytree(args.project_directory, args.output_directory)
+
+    # Extract all of the kernel parameter and template type information.
+    if args.add_static_casts:
+        KernelTemplateParams = {}
+        for (dirpath, _dirnames, filenames) in os.walk(args.output_directory):
+            for filename in filenames:
+                if filename_ends_with_extension(filename, args.extensions) and inside_included_directories(dirpath, args.output_directory, args.include_dirs):
+                    the_file = os.sep.join([dirpath, filename])
+
+                    # Store param information inside KernelTemplateParams
+                    get_kernel_template_params(the_file, KernelTemplateParams)
+
+    # Open YAML file with disable information.
+    if args.yaml_settings != "":
+        with openf(args.yaml_settings, "r") as lines:
+            yaml_data = yaml.load(lines)
+
+        # Disable functions in certain files according to YAML description
+        for disable_info in yaml_data["disabled_functions"]:
+            filepath = os.path.join(args.output_directory, disable_info["path"])
+            if "functions" in disable_info:
+                functions = disable_info["functions"]
+            else:
+                functions = []
+
+            if "non_hip_functions" in disable_info:
+                non_hip_functions = disable_info["non_hip_functions"]
+            else:
+                non_hip_functions = []
+
+            if "non_device_functions" in disable_info:
+                not_on_device_functions = disable_info["non_device_functions"]
+            else:
+                not_on_device_functions = []
+
+            with openf(filepath, "r+") as f:
+                txt = f.read()
+                for func in functions:
+                    # TODO - Find fix assertions in HIP for device code.
+                    txt = disable_function(txt, func, disablefuncmode.ASSERT)
+
+                for func in non_hip_functions:
+                    # Disable this function on HIP stack
+                    txt = disable_function(txt, func, disablefuncmode.HCC_MACRO)
+
+                for func in not_on_device_functions:
+                    # Disable this function when compiling on Device
+                    txt = disable_function(txt, func, disablefuncmode.DEVICE_MACRO)
+
+                f.seek(0)
+                f.write(txt)
+                f.truncate()
+                f.close()
```

Comment:
You don't have to close the file; the context manager will do it for you.

Reply:
done